src/rules/trailingCommaRule.ts

```diff
@@ -20,11 +20,98 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+type OptionValue = "always" | "never" | "ignore";
+
+interface CustomOptionValue {
+    arrays?: OptionValue;
+    exports?: OptionValue;
+    functions?: OptionValue;
+    imports?: OptionValue;
+    objects?: OptionValue;
+    typeLiterals?: OptionValue;
+}
+
 interface Options {
-    multiline?: "always" | "never";
-    singleline?: "always" | "never";
+    multiline?: OptionValue | CustomOptionValue;
```

Comment:
You don't need `OptionValue` here, as you normalize it anyway. You don't need to cast when using it below